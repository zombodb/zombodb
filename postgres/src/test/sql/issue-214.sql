CREATE TABLE issue_214 (id SERIAL PRIMARY KEY, name TEXT);
CREATE INDEX idx_zdb_foo ON issue_214 USING zombodb(zdb('issue_214', ctid), zdb(issue_214)) WITH (url="http://localhost:9200/");
INSERT INTO issue_214 (name) VALUES ('a!a');

SELECT * FROM zdb_dump_query('issue_214', '_all:a\!*');
SELECT assert(zdb_dump_query('issue_214', '_all:a\!*'), zdb_dump_query('issue_214', '_all:"a!*"'), '_all: wildcard escaped v/s not');
SELECT * FROM issue_214 WHERE zdb('issue_214', ctid) ==> '_all:a\!*';


SELECT * FROM zdb_dump_query('issue_214', 'name:a\!*');
SELECT assert(zdb_dump_query('issue_214', 'name:a\!*'), zdb_dump_query('issue_214', 'name:"a!*"'), 'name: wildcard escaped v/s not');
SELECT * FROM issue_214 WHERE zdb('issue_214', ctid) ==> 'name:a\!*';

DROP TABLE issue_214;
