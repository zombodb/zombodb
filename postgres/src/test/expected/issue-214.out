CREATE TABLE issue_214 (id SERIAL PRIMARY KEY, name TEXT);
CREATE INDEX idx_zdb_foo ON issue_214 USING zombodb(zdb('issue_214', ctid), zdb(issue_214)) WITH (url="http://localhost:9200/");
INSERT INTO issue_214 (name) VALUES ('a!a');
SELECT * FROM zdb_dump_query('issue_214', '_all:a\!*');
      zdb_dump_query      
--------------------------
 {                       +
   "span_near" : {       +
     "clauses" : [ {     +
       "span_term" : {   +
         "_all" : {      +
           "value" : "a" +
         }               +
       }                 +
     }, {                +
       "span_multi" : {  +
         "match" : {     +
           "wildcard" : {+
             "_all" : "*"+
           }             +
         }               +
       }                 +
     } ],                +
     "slop" : 0,         +
     "in_order" : true   +
   }                     +
 }
(1 row)

SELECT assert(zdb_dump_query('issue_214', '_all:a\!*'), zdb_dump_query('issue_214', '_all:"a!*"'), '_all: wildcard escaped v/s not');
 assert 
--------
 t
(1 row)

SELECT * FROM issue_214 WHERE zdb('issue_214', ctid) ==> '_all:a\!*';
 id | name 
----+------
  1 | a!a
(1 row)

SELECT * FROM zdb_dump_query('issue_214', 'name:a\!*');
  zdb_dump_query   
-------------------
 {                +
   "prefix" : {   +
     "name" : "a!"+
   }              +
 }
(1 row)

SELECT assert(zdb_dump_query('issue_214', 'name:a\!*'), zdb_dump_query('issue_214', 'name:"a!*"'), 'name: wildcard escaped v/s not');
 assert 
--------
 t
(1 row)

SELECT * FROM issue_214 WHERE zdb('issue_214', ctid) ==> 'name:a\!*';
 id | name 
----+------
  1 | a!a
(1 row)

DROP TABLE issue_214;
